lib/logstash/inputs/xmpp.rb

```diff
@@ -67,14 +80,32 @@ def run(queue)
       if msg.body != nil
         @codec.decode(msg.body) do |event|
           decorate(event)
-          # Maybe "from" should just be a hash: 
+          # Maybe "from" should just be a hash:
           # { "node" => ..., "domain" => ..., "resource" => ... }
           event["from"] = "#{msg.from.node}@#{msg.from.domain}/#{msg.from.resource}"
           queue << event
         end
       end
     end # @client.add_message_callback
+
+    # [GUY] the two clients: muc and client maintain their own threads
+    # and will call the blocks in those threads so this one should sleep
+    # this is how the xmpp4r examples do it (not saying it is correct)
     sleep
+
+    @client.close
   end # def run
 
+  def stop
+    super
```

Comment:
no need to call `super` per he latest changes which occurred on the `3895` logstash PR